google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java

```diff
@@ -179,12 +187,26 @@ public boolean value(VirtualFile file) {
         GctBundle.message("appengine.dockerfile.location.browse.button"),
         null,
         project,
-        FileChooserDescriptorFactory.createSingleFileDescriptor());
+        FileChooserDescriptorFactory
+            .createSingleFileDescriptor().withFileFilter(new Condition<VirtualFile>() {
+          @Override
+          public boolean value(VirtualFile file) {
+            return Comparing.equal(file.getName(), "Dockerfile", false);
+          }
+        })
+    );
     appYamlPathField.addBrowseFolderListener(
         GctBundle.message("appengine.appyaml.location.browse.window.title"),
         null,
         project,
-        FileChooserDescriptorFactory.createSingleFileDescriptor());
+        FileChooserDescriptorFactory
+            .createSingleFileDescriptor().withFileFilter(new Condition<VirtualFile>() {
+          @Override
+          public boolean value(VirtualFile file) {
+            return Comparing.equal(file.getName(), "app.yaml", false);
```

Comment:
Messing around I do seem to be able to call `gcloud preview app deploy bla.yaml`..  I'm just wondering if users might do things like app.qa.yaml, app.prod.yaml, and even Dockerfile.qa, and then get any names right (if they need to be) during staging.